pkg/splunk/enterprise/licensemaster.go

```diff
@@ -74,6 +75,16 @@ func ApplyLicenseMaster(client splcommon.ControllerClient, cr *enterpriseApi.Lic
 		if err != nil {
 			return result, err
 		}
+
+		// If this is the last of its kind getting deleted,
+		// remove the entry for this CR type from configMap
+		if len(cr.Spec.AppFrameworkConfig.AppSources) != 0 {
+			err = UpdateOrRemoveEntryFromConfigMap(client, cr, SplunkLicenseMaster)
+			if err != nil {
+				return result, err
+			}
+		}
```

Comment:
I think the best place to address this is, as part of the `DeleteOwnerReferencesForResources`.  Also, looking at `UpdateOrRemoveEntryFromConfigMap`, since we don't have exact mapping at CR level, how are we sure that the refCount is already not decremented for this CR?

Reply:
Each CR is responsible for decrementing its own refCount in its own reconcile.